cargo/cargo_build_script.bzl

```diff
@@ -316,32 +303,43 @@ def cargo_build_script(
     build script in addition to the file generated by it.
 
     Args:
-        name (str): The target name for the underlying rule
-        crate_name (str, optional): Name of the crate associated with this build script target.
+        name (str): The target name for the underlying rule. This should be the name of the crate being compiled, optionally with a suffix of _build_script.
         crate_features (list, optional): A list of features to enable for the build script.
         version (str, optional): The semantic version (semver) of the crate.
-        deps (list, optional): The dependencies of the crate defined by `crate_name`.
+        deps (list, optional): The dependencies of the crate.
         build_script_env (dict, optional): Environment variables for build scripts.
         data (list, optional): Files or tools needed by the build script.
         links (str, optional): Name of the native library this crate links against.
+        rustc_env (dict, optional): Environment variables to set in rustc when compiling the build script.
```

Comment:
This doesn't seem to be mentioned in the PR description -- can you add a note there?

Also, what's the reason for adding this here? Is this just for commonality with the other rules?

Reply:
This isn't actually a change in behaviour at all - the `kwargs` are passed through to the underlying `rust_binary` already, this just names one of them so that we can operate on it with a name, rather than as `kwargs.get("rustc_env", {})`. The equivalent code without this would be something like:

```python
if "rustc_env" not in kwargs:
    kwargs["rustc_env"] = {}
rustc_env = dict(kwargs["rustc_env"])
if "CARGO_PKG_NAME" not in rustc_env:
    rustc_env["CARGO_PKG_NAME"] = _name_to_pkg_name(name)
if "CARGO_CRATE_NAME" not in rustc_env:
    rustc_env["CARGO_CRATE_NAME"] = name_to_crate_name(_name_to_pkg_name(name))
```

which seems a little less tidy, but maybe is more clear about what's going on? Either way, the intent I'm trying to convey is "We always pass through a `rustc_env` to the `rust_binary`, you can add to it if you want"...

Reply:
Got it -- that makes total sense.

So I think the only issue here is my lack of understanding, and there's nothing to be done here.